packages/dashboard-frontend/src/services/workspace-client/devworkspace/__tests__/devWorkspaceClient.create.spec.ts

```diff
@@ -38,74 +41,234 @@ describe('DevWorkspace client, create', () => {
-            'che.eclipse.org/last-updated-timestamp': timestampNew,
-          },
+        [],
+        undefined,
+        undefined,
+        editorId,
+        {},
+      );
+
+      expect(spyCreateWorkspace).toBeCalledWith(
+        expect.objectContaining({
+          metadata: expect.objectContaining({
+            annotations: expect.objectContaining({
+              'che.eclipse.org/che-editor': editorId,
+            }),
+          }),
         }),
-      }),
-    );
+      );
+    });
   });
 
-  it('should add editor annotation', async () => {
-    const namespace = 'che';
-    const name = 'wksp-test';
-    const testDevfile: devfileApi.Devfile = {
-      schemaVersion: '2.1.0',
-      metadata: {
+  describe('Create from resources', () => {
+    let testDevWorkspace: devfileApi.DevWorkspace;
+    let testDevWorkspaceTemplate: devfileApi.DevWorkspaceTemplate;
+
+    beforeEach(() => {
+      testDevWorkspace = {
+        apiVersion: 'workspace.devfile.io/v1alpha2',
+        kind: 'DevWorkspace',
+        metadata: {
+          namespace,
+          name,
+          uid: '1234567890',
+          labels: {},
+        },
+        spec: {
+          started: false,
+          template: {},
+        },
+      };
+      testDevWorkspaceTemplate = {
+        apiVersion: 'workspace.devfile.io/v1alpha2',
+        kind: 'DevWorkspaceTemplate',
+        metadata: {
+          namespace,
+          name,
+          annotations: {},
+        },
+        spec: {
+          components: [
+            {
+              name: 'test-component-name',
+              container: {
+                image: 'test.image',
+              },
+              attributes: {},
+            },
+          ],
+        },
+      };
+    });
+
+    it('should add annotation of last update time', async () => {
+      const testWorkspace = new DevWorkspaceBuilder()
+        .withMetadata({
+          name,
+          namespace,
+        })
+        .build();
+
+      const spyCreateWorkspace = jest
+        .spyOn(DwApi, 'createWorkspace')
+        .mockResolvedValueOnce(testDevWorkspace);
+      jest.spyOn(DwtApi, 'createTemplate').mockResolvedValueOnce(testDevWorkspaceTemplate);
+      jest.spyOn(DwApi, 'patchWorkspace').mockResolvedValueOnce(testWorkspace);
+
+      await client.createFromResources(
         namespace,
-        name,
-      },
-    };
-    const testWorkspace = new DevWorkspaceBuilder()
-      .withMetadata({
-        name,
+        testDevWorkspace,
+        testDevWorkspaceTemplate,
+        undefined,
+        undefined,
+        undefined,
+      );
+
+      expect(spyCreateWorkspace).toBeCalledWith(
+        expect.objectContaining({
+          metadata: expect.objectContaining({
+            annotations: {
+              'che.eclipse.org/last-updated-timestamp': timestampNew,
+            },
+          }),
+        }),
+      );
+    });
+
+    it('should add pluginRegistry and dashboard URLs as environment variables', async () => {
```

Comment:
Please review this test because it seems to be identical to the previous one.

Reply:
I have fixed it.